NewForce-Cohort-6/tabloidfullstack-stuffing
# Nt/13 view all tags

[Trello ticket 13-- View All Tags](https://trello.com/c/FolQ4LjI/13-13-view-all-tags)

Creates:
Client > src > components > tags
tags >TagList.js
tags > TagListItem.js
Managers >TagManager.js

Tabloid > Models > Tag.cs
Tabloid > Repositories > TagRepository.cs //GetAll()
Tabloid > Repositories > ITagRepository.cs //List<Tag> GetAll()
Tabloid > Controllers > TagController.cs

Changes
Tabloid > Startup.cs (adds services.AddTransient<ITagRepository, TagRepository>(); )
Tabloid > Client > src > nav > Header.js (adds Tag Management link)

Testing Instructions
1. `git add` and `git commit` whatever you're working on on your branch

2. `git checkout main`

3. `git fetch --a` (or `git fetch --all`)

4. `git checkout NT/13-View-All-Tags` to switch to my branch.

5. If you don't see the most recent version of the files, try `git pull origin NT/13-View-All-Tags`

6. If you haven't created the Tabloid database, please do so.

7. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/01_Tabloid_Create_DB.sql)

8. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-stuffing/blob/main/SQL/02_Tabloid_Seed_Data.sql)

9. In your terminal, navigate to the client folder (something like cd workspace/Tabloid/Tabloid/Client)

10. If you have not run the command `npm install` in your terminal at this folder level, do so now.

11. Go back to Visual Studio. Execute the backend of Tabloid from inside of Visual Studio by clicking on the solid green arrow.

12.  A browser running Swagger should pop up.

13. In your terminal, run the command `npm start`

14. A browser tab running Tabloid should appear.

15. In the browser, please log in as [foo@bar.com](mailto:foo@bar.com) (admin profile).

16. Use the Navbar to navigate to the Tag Management page.

17. You should see a table with one column labeled "Tags:"

18. You should see the following tags in alphabetical order:
- C#
- Funny
- JavaScript
- Scary

19.  If you have added any additional tags, they will also appear in the correct order alphabetically.

20. If you see an alphabetical list of Tags as described in 18, the feature works.

Checklist:
 [x] My code follows the style guidelines of this project
 [x] I have performed a self-review of my code
 [x] I have commented my code, particularly in hard-to-understand areas
 [x] My changes generate no new warnings or errors
 [x] I have added test instructions that prove my fix is effective or that my feature works
